superflore/generators/bitbake/yocto_recipe.py

```diff
@@ -117,68 +225,570 @@ def get_src_location(self):
+                if not results:
+                    yoctoRecipe.rosdep_cache[dep] = []
+                    continue
+                for res in results:
+                    recipe = self.convert_to_oe_name(res, is_native)
+                    dependencies.add(recipe)
+                    system_dependencies.add(self.convert_to_oe_name(res))
+                    yoctoRecipe.rosdep_cache[dep].add(res)
+                    info('External dependency add: ' + recipe)
+            except UnresolvedDependency:
+                info('Unresolved dependency: ' + dep)
+                if dep in yoctoRecipe.rosdep_cache:
+                    cached_deps = yoctoRecipe.rosdep_cache[dep]
+                    if cached_deps == set(['null']):
+                        system_dependencies.add(dep)
+                        recipe = dep + self.get_native_suffix(is_native)
+                        dependencies.add(recipe)
+                        msg = 'Failed to resolve (cached):'
+                        warn('{0} {1}: {2}'.format(msg, dep, recipe))
+                    elif cached_deps:
+                        system_dependencies |= cached_deps
+                        for d in cached_deps:
+                            recipe = self.convert_to_oe_name(d, is_native)
+                            dependencies.add(recipe)
+                            msg = 'Resolved in OpenEmbedded (cached):'
+                            info('{0} {1}: {2}'.format(msg, dep, recipe))
+                    continue
+                oe_query = OpenEmbeddedLayersDB()
+                oe_query.query_recipe(self.convert_to_oe_name(dep))
+                if oe_query.exists():
+                    recipe = self.convert_to_oe_name(oe_query.name, is_native)
+                    dependencies.add(recipe)
+                    oe_name = self.convert_to_oe_name(oe_query.name)
+                    system_dependencies.add(oe_name)
+                    yoctoRecipe.rosdep_cache[dep].add(oe_name)
+                    info('Resolved in OpenEmbedded: ' + dep + ' as ' +
+                         oe_query.name + ' in ' + oe_query.layer +
+                         ' as recipe ' + recipe)
+                else:
+                    recipe = dep + self.get_native_suffix(is_native)
+                    dependencies.add(recipe)
+                    system_dependencies.add(dep)
+                    yoctoRecipe.rosdep_cache[dep].add('null')
+                    warn('Failed to resolve fully: ' + dep)
+        return dependencies, system_dependencies
+
+    def get_recipe_text(self, distributor):
         """
         Generate the Yocto Recipe, given the distributor line
         and the license text.
         """
-        ret = "# Copyright " + strftime("%Y", gmtime()) + " "
-        ret += distributor + "\n"
-        ret += '# Distributed under the terms of the ' + license_text
-        ret += ' license\n\n'
-
+        ret = "# Generated by superflore -- DO NOT EDIT\n#\n"
```

Comment:
It might be useful to add a timestamp to this

Reply:
> @andre-rosa I'd really like to see a bit more coverage for `oe_query.py`.
> 
> ```
> superflore/generators/bitbake/oe_query.py            104     91    12%   24-44, 47-49, 53-58, 61-96, 99-121, 124, 127-136, 140-150, 154
> ```
> 
> This PR reduces the total coverage of the repository by around 10%, which makes sense (since it's adding so many things). Do you think you could add some tests for that file?

Definitely; will do it.

Reply:
Adding a timestamp would ream that every file will different every time superflore it run, making it difficult to find the "real" changes. There's a timestamp in conf/ros-distro/generated-ros-distro.inc: ROS_SUPERFLORE_GENERATION_DATETIME .

Reply:
Right, that's true. My bad.